src/bench/connection.rs

```diff
@@ -118,6 +121,12 @@ impl Connection {
         let mut incoming_done = false;
         let mut acks_count = 0;
         let mut incoming_count = 0;
+        let mut hist = Histogram::<u64>::new(4).unwrap();
+
+        // maps to record Publish and PubAck of messages. PKID acts as key
+        let mut pkids_publish: BTreeMap<u16, std::time::Instant> = BTreeMap::new();
+        let mut vector: VecDeque<std::time::Instant> = VecDeque::with_capacity(self.config.max_inflight as usize);
```

Comment:
`push_back` and `pop_front` below are not correct because broker can be unordered. This queue should be `Vec<Option<Duration>>` and initialized with all `None`s. An outgoing publish will replace None with `Some(Duration)` for at index = pkid.

Reply:
Ah I didn't know that the acks could be unordered.